src/Components/Table/Table.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  DataGrid,
+  GridColDef,
+  GridEventListener,
+  GridSortModel,
+} from "@mui/x-data-grid";
+import { Paper } from "@mui/material";
+import { TableProps } from "./types";
+import CustomFooter from "./CustomFooter";
+import { sortType } from "./consts";
+
+const Table = ({
+  cols,
+  rows,
+  onRowClick: onClick,
+  style,
+  headerClassName,
+  headerTypo,
+  disableColumnMenu = true,
+  disableColumnSorting = true,
+  sortBy = { col: "", order: sortType.ASC },
+}: TableProps) => {
+  const columns: GridColDef[] = cols.map((col) => ({
+    field: col.field,
+    headerName: col.title,
+    headerClassName: headerClassName,
+    width: col.width || 150,
+    minWidth: col.minWidth || 0,
+    flex: col.flex || 0,
+    renderCell: col.renderCell || ((params) => params.value),
+    renderHeader: () => headerTypo(col.title),
+  }));
+
+  const sortModel: GridSortModel = sortBy
+    ? [{ field: sortBy.col, sort: sortBy.order }]
+    : [];
+  const handleRowClick: GridEventListener<"rowClick"> = (params, event) => {
+    onClick(params.row.name);
```

Comment:
why don;y tou just pass the onClick prop as you get it directly to the table?
like so: `onRowClick={handleRowClick}` where `handleRowClick` is the prop directly, not the function that you implemented above.